### ISSUE TITLE
Crossfade with ALSA device results in failure to play on song switch

### ISSUE THREAD

**Issue by rijnhard** [jonaski/strawberry#48]
To reproduce use the following settings (use own device) and start playing one song then double click to start playing a second song (without stopping the first song). 

* Backend
** Backend: GStreamer
** Output: to sound card via ALSA
** Device: Meridian Explorer USB DAC
* Playback
** Cross-fade when changing tracks manually: true

Logs:
```
10:02:05.748 ERROR GstEnginePipeline:575            ErrorMessageReceived 2 "gstalsasink.c(858): gst_alsasink_open (): /GstBin:audiobin/GstAlsaSink:alsasink-16:\nDevice 'hw:1,0' is busy"
10:02:05.791 ERROR GstEngine:553                    Gstreamer error: 4227 4 "Could not open audio device for playback. Device is being used by another application."
__logging_message__10:02:05.793 WARN  unknown                          QTimeLine::start: already running
10:02:09.661 ERROR logging:60                       Source ID 34 was not found when attempting to remove it
```

This can be quite a pita, I'd suggest that for a phase 1 you just ignore the crossfade settings for alsa devices, because I can imagine that mixing the audio will be a lot of work.

**Comment by alekksander** [comment]
please correct me if i'm wrong, but mixing is a job for alsa, not the player itself. use plughw and proper alsa configuration to make it work. cross-fading is not suppose to work with direct hw. therefore i think Your statement for making fading options unavailable when playing with alsa/hw is a good way to go. let's see what the dev has to say about that.

**Comment by jonaski** [comment]
It's true that fading won't work properly if you select a alsa device, as it will open two streams at the same time, in same way that you can't use sound in two applications at the same time. It works if you use alsa with automatically select, since that will use dmix and resample.
That's the reason I made the fading disabled by default. However I agree that the fading option should be disabled when a alsa device is selected, as otherwise it seems broken. I actually intended to do it, but never got to it. I will fix it now.
It seems like plughw works the same way as hw how, and does not resample either.

**Comment by jonaski** [comment]
Fixed in 513aafc

**Comment by rijnhard** [comment]
Can confirm it greys out
